idb/postgres/postgres_integration_test.go

```diff
@@ -1783,3 +1793,93 @@ func TestKeytypeDoNotResetReceiver(t *testing.T) {
 	assertKeytype(t, db, test.AccountA, &keytype)
 	assertKeytype(t, db, test.AccountB, &keytype)
 }
+
+// Test that if information in `txn` and `txn_participation` tables is ahead of
+// the current round, AddBlock() still runs successfully.
+func TestAddBlockTxnTxnParticipationAhead(t *testing.T) {
```

Comment:
How about the "Behind" case? Previously txn and txn_participation were always added first, so they were kind of always ahead.

Reply:
Behind is not possible. We commit the main transaction only when the two auxiliary transactions are committed.